module-network.tf

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) 2017, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+data "oci_core_vcns" "all" {
+  count          = var.create_vcn ? 0 : 1
+  compartment_id = local.compartment_id
+  display_name   = var.vcn_name
+  state          = "AVAILABLE"
+}
+
+data "oci_core_vcn" "oke" {
+  count  = var.create_vcn ? 0 : 1
+  vcn_id = var.vcn_id
+}
+
+data "oci_core_nat_gateways" "all" {
+  compartment_id = local.compartment_id
+  display_name   = var.nat_gateway_display_name
```

Comment:
as for vcn, `display_name` is not unique

Reply:
Responded in other comments - we can remove this if we decide to make strict i.e. must provide vcn_id explicitly if create is disabled.

Reply:
Removed as with VCN fallback.